modules/v1/workspaces/resources/ArticleFileResource.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+
+namespace app\modules\v1\workspaces\resources;
+
+
+use app\modules\v1\users\resources\UserResource;
+use app\modules\v1\workspaces\models\ArticleFile;
+use Yii;
+use yii\db\ActiveQuery;
+
+/**
+ * Class ArticleFileResource
+ *
+ * @package app\modules\v1\workspaces\resources
+ */
+class ArticleFileResource extends ArticleFile
+{
+    public function fields()
+    {
+        return [
+            'id',
+            'name',
+            'label',
+            'size',
+            'to_process',
+            'processing',
+            'mime' => function () {
+                $mime = explode('/', $this->mime)[0];
+                return $mime == 'video' ? $mime : $this->mime;
+            },
+            'path' => function () {
+                if (explode('/', $this->mime)[0] == 'video') {
```

Comment:
@skavt You can create getPubicUrl method and move logic there and use it here simply like `$this->getPublicUrl()`